### ISSUE TITLE
ring uberwar compiles some libs into WEB-INF/lib causing problems on windows

### ISSUE THREAD

**Issue by yatesco** [weavejester/lein-ring#120]
TLDR: lein ring uberwar seems to explode and compile into WEB-INF/classes a number of dependencies as well as include them in WEB-INF/lib.

After doing lein ring uberwar I see all the expected JARs I reference in WEB-INF/lib which is great.  I also see the deconstructed and compiled version in WEB-INF/classes for some of them.

This seems unnecessary and causes problems when the names of functions differ in case only on Windows.  Specifically camel-snake-kebab has two functions that only differ in case.  This causes deploying the WAR in windows to barf with (for example): `NoClassDefFoundError:camel_snake_kebab/core$__GT_CamelCaseString(wrong name: camel_snake_kebab/core$__GT_camelCaseString)`.  (more info at https://github.com/qerub/camel-snake-kebab/issues/15)

My project.clj:

```
(defproject X.X/X "0.1.0-SNAPSHOT"
  :description "XXX"
  :url "XXX"
  :dependencies [[org.clojure/clojure "1.6.0"]
                 ;; compojure 1.1.6 onwards no longer comes with
                 ;; servlet API...
                 [compojure "1.1.8"]                 
                 [ring/ring-json "0.3.1"]
                 [clj-time "0.8.0"]
                 [org.clojure/java.jdbc "0.3.5"]
                 ;; do not upgrade jtds until everything uses jdk 1.7
                 [net.sourceforge.jtds/jtds "1.2.7"]
                 [com.mchange/c3p0 "0.9.2.1"]
                 [environ "0.5.0"]
                 [org.clojure/tools.logging "0.3.0"]
                 [ch.qos.logback/logback-classic "1.1.2"]
                 [honeysql "0.4.3"]
                 [clj-http "0.9.2"]
                 [org.apache.poi/poi "3.9"]
                 [org.apache.poi/poi-scratchpad "3.9"]
                 [org.apache.xmlgraphics/batik-transcoder "1.7"]
                 [org.apache.xmlgraphics/batik-codec "1.7"]
                 [camel-snake-kebab "0.2.1"]
                 ;; needed for those using commons logging
                 [org.slf4j/jcl-over-slf4j "1.7.7"]]
  ;; exclude commons-logging as it causes classpath issues because it isn't set up correctly.
  ;; it is pulled in by clj-http but it is excluded here for safety.
  :exclusions [commons-logging/commons-logging]
  :plugins [[lein-ring "0.8.11"]
            [lein-marginalia "0.7.1"]
            [lein-shell "0.4.0"]
            [lein-cucumber "1.0.2"]
            [lein-midje "3.1.3"]
            [lein-deps-tree "0.1.2"]
            [lein-environ "0.5.0"]
            [lein-ancient "0.5.5"]]
  :ring {:handler health.handler/app}
  :profiles {:dev {:dependencies [[ring-mock "0.1.5"]
                                  [midje "1.6.3"]
                                  [org.clojure/tools.trace "0.7.8"]
                                  [hsqldb/hsqldb "1.8.0.10"]
                                  ;; compojure 1.1.6 doesn't come with servlet
                                  ;; api anymore.
                                  [javax.servlet/servlet-api "2.5"]]
                   :env {:analytocs-skip-populating-date-dimension-tables true}}
             :uberjar {:aot false}}
  :cucumber-feature-paths ["test/features/"]
  :aliases {"cucumber!" ["cucumber" "-f" "json:target/cucumber-test-report.json" "-f" "junit:target/cucumber-test-report.xml" "-f" "pretty"]
            "phantomjs!" ["shell" "phantomjs" "resources/public/lib/run-jasmine-1.9.1.js" "resources/public/test.html"]
            "war!" ["ring" "uberwar" "health.war"]
            "web!" ["with-profile" "web" "ring" "server-headless"]
            ;; run all the tests (except for cucumber as it isn't used yet)
            "all-tests!" ["do" "midje" ":config" "jenkins-midje.clj," "phantomjs!"]
            "marg!" ["do" "marg" "-d" "target/doc"]
            "full-build!" ["do" "clean," "all-tests!," "war!," "marg!"]})
```

This causes `lein war!` to generate the following:

![snapshot1](https://cloud.githubusercontent.com/assets/181469/4077890/0ca324b4-2ec7-11e4-8cf6-504d13e02026.png)
